File: AdPieSample/AdPieSample/Source/BannerAdViewController.m

#import "BannerAdViewController.h"
#import "UIViewController+Toast.h"
#import <AdPieSDK/AdPieSDK.h>

@interface BannerAdViewController () <APAdViewDelegate>
@property (nonatomic, copy) NSString *slotId;
@property (nonatomic, strong) APAdView *bannerView;
@end

@implementation BannerAdViewController

- (instancetype)initWithSlotId:(NSString *)slotId {
    self = [super initWithNibName:nil bundle:nil];
    if (self) {
        _slotId = [slotId copy];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor whiteColor];
    
    [self setupUI];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    // Load Ad
    [self.bannerView load];
}


- (void)setupUI {
    // Banner View
    self.bannerView = [[APAdView alloc] init];
    self.bannerView.slotId = self.slotId;
    self.bannerView.delegate = self;
    self.bannerView.rootViewController = self;
    // AutoLayout
    [self.view addSubview:self.bannerView];
    self.bannerView.translatesAutoresizingMaskIntoConstraints = NO;
    [NSLayoutConstraint activateConstraints:@[
        [self.bannerView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
        [self.bannerView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor constant:10],
        [self.bannerView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor constant:-10],
        [self.bannerView.heightAnchor constraintEqualToConstant:50]
    ]];
    self.bannerView.onPaidEvent = ^(double ecpm) {
        NSLog(@"onPaidEvent, ecpm: %f", ecpm);
    };
}

#pragma mark APAdView delegates
- (void)adViewDidLoadAd:(APAdView *)view {
    [self showToastWithMessage:[NSString stringWithFormat:@"%s",__PRETTY_FUNCTION__]];
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

- (void)adViewDidFailToLoadAd:(APAdView *)view withError:(NSError *)error {
    [self showToastWithMessage:[NSString stringWithFormat:@"%s",__PRETTY_FUNCTION__]];
    NSString * errorMsg = [NSString stringWithFormat:@"Error (code : %d, message : %@, date : %@)",
                           (int)[error code],
                           [error localizedDescription],
                           [[NSDate date] descriptionWithLocale:[NSLocale systemLocale]]];
    NSLog(@"%s, %@", __PRETTY_FUNCTION__, errorMsg);
}

- (void)adViewWillLeaveApplication:(APAdView *)view {
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

@end

File: AdPieSwiftSample/AdPieSwiftSample/Source/BannerAdViewController.swift
import UIKit
import AdPieSDK

class BannerAdViewController: UIViewController, APAdViewDelegate {
    
    private var slotId: String?
    private var bannerView: APAdView!
    
    init(slotId: String?) {
        self.slotId = slotId
        super.init(nibName: nil, bundle: nil)
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = .white
        
        setupUI()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        // Load Ad
        bannerView.load()
    }
    
    private func setupUI() {
        // Banner View
        bannerView = APAdView()
        bannerView.slotId = slotId
        bannerView.delegate = self
        bannerView.rootViewController = self
        
        // AutoLayout
        view.addSubview(bannerView)
        bannerView.translatesAutoresizingMaskIntoConstraints = false
        
        NSLayoutConstraint.activate([
            bannerView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
            bannerView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 10),
            bannerView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -10),
            bannerView.heightAnchor.constraint(equalToConstant: 50)
        ])
        
        bannerView.onPaidEvent = { ecpm in
            print("onPaidEvent, ecpm: \(ecpm)")
        }
    }
    
    // MARK: - APAdView delegates
    
    func adViewDidLoadAd(_ view: APAdView!) {
        showToast(message: "\(#function)")
        print("\(#function)")
    }
    
    func adViewDidFail(toLoadAd view: APAdView!, withError error: Error!) {
        showToast(message: "\(#function)")
        let nsError = error as NSError
        let errorMsg = String(format: "Error (code : %d, message : %@, date : %@)",
                              Int32(nsError.code),
                              nsError.localizedDescription,
                              Date().description(with: Locale.current))
        print("\(#function), \(errorMsg)")
    }
    
    func adViewWillLeaveApplication(_ view: APAdView!) {
        print("\(#function)")
    }
}
